### PR TITLE
refactor: enable strict mode and update accordingly

### PR DESCRIPTION
**Pull request by rafaelss95**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
No strict mode.

Issue Number: N/A

## What is the new behavior?
As per https://github.com/ngneat/bind-query-params/pull/5#issuecomment-778736574, it enables the strict mode.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -140,11 +156,16 @@
...
         }
       }
     }
   },
-  "defaultProject": "bind-query-params"
-}
+  "defaultProject": "bind-query-params",
+  "cli": {
+    "analytics": false
+  }
```
Comment (rafaelss95) [inline angular.json:170]: These changes are generated by the `@angular/cli`. When you run `ng serve`, it asks for report analytics.

File: projects/ngneat/bind-query-params/src/lib/BindQueryParamsManager.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export class BindQueryParamsManager<T = any> {
     this.updateControl(this.defs, { emitEvent: true }, (def) => def.strategy === 'twoWay');
 
     const controls = this.defs.map((def) => {
-      return this.group.get(def.path).valueChanges.pipe(
+      return this.group.get(def.path)!.valueChanges.pipe(
```
Comment (rafaelss95) [inline projects/ngneat/bind-query-params/src/lib/BindQueryParamsManager.ts:34]: This case can be problematic, but I don't know if there's a better option to handle this.

File: projects/ngneat/bind-query-params/src/lib/BindQueryParamsManager.ts [hunk 3]
```diff
@@ -115,15 +119,16 @@ export class BindQueryParamsManager<T = any> {
     updatePredicate = (_: QueryParamDef) => true
   ) {
     const queryParams = new URLSearchParams(this.options.windowRef.location.search);
-    let value = {};
+    let value: Partial<T> = {};
 
     for (const def of defs) {
       if (updatePredicate(def)) {
         const { queryKey } = def;
+        const queryDef = queryParams.get(queryKey);
```
Comment (NetanelBasal) [inline projects/ngneat/bind-query-params/src/lib/BindQueryParamsManager.ts:127]: why did you change it?

Comment (rafaelss95) [reply projects/ngneat/bind-query-params/src/lib/BindQueryParamsManager.ts:127]: Due to these errors:

Captura de Tela 2021-02-16 às 10 03 14
Captura de Tela 2021-02-16 às 10 03 29

With the `.has(...)` checkage TS isn't able to know that `.get(...)` is not null. So, in order to solve this problem, I moved it a variable and use it later. Ofc, we can get rid of this using `!.`, but in general it's better to avoid the non-null-assertion operator.
